File: 36-valid-sudoku/valid-sudoku.cpp
#include<vector>
#include<set>
#include<map>
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        vector<set<char>> validRow;
        vector<set<char>> validCol;
        map<int, set<char>> validBox;
        for(int i = 0; i < board.size(); i++){
            set<char> temp;
            validRow.push_back(temp);
            validCol.push_back(temp);
            for(int j = 0; j < board.size(); j++){
                int key = (i/3)*10 + (j/3);
                validBox[key] = temp;
            }
        }
        for(int i = 0; i < board.size(); i++){
            for(int j = 0; j < board.size(); j++){
                int key = (i/3)*10 + (j/3);
                if (board[i][j] == '.') continue;
                if (validRow[i].find(board[i][j]) != validRow[i].end() or validCol[j].find(board[i][j]) != validCol[j].end() or validBox[key].find(board[i][j]) != validBox[key].end()){

                    return false;
                }
                validRow[i].insert(board[i][j]);
                validCol[j].insert(board[i][j]);
                validBox[key].insert(board[i][j]);
            }
        }
        for(set<char> i: validRow){
            for(char j: i){
                
            }
        }
        return true;
    }
};

File: 36-valid-sudoku/valid-sudoku.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Vector;

class Solution {
    public boolean isValidSudoku(char[][] board) {
        Vector<HashSet<Character>> validRow = new Vector<HashSet<Character>>();
        Vector<HashSet<Character>> validCol = new Vector<HashSet<Character>>();
        HashMap<Integer, HashSet<Character>> validBox  = new HashMap<Integer, HashSet<Character>>();

        for(int i = 0; i < board.length; i++){
            HashSet<Character> temp = new HashSet<Character>();
            validRow.add(temp);
            HashSet<Character> temp2 = new HashSet<Character>();
            validCol.add(temp2);
            for(int j = 0; j < board.length; j++){
                int key = (i/3)*10 + (j/3);
                HashSet<Character> temp3 = new HashSet<Character>();
                validBox.put(key, temp3);
            }
            
        }

        for(int i = 0; i < board.length; i++){
            for(int j = 0; j < board.length; j++){
                int key = (i/3)*10 + (j/3);
                if(board[i][j] == '.') continue;
                if( validRow.get(i).contains(board[i][j]) ||  validCol.get(j).contains(board[i][j]) || validBox.get(key).contains(board[i][j])){
                    return false;
                } 
                validRow.get(i).add(board[i][j]);
                validCol.get(j).add(board[i][j]);
                validBox.get(key).add(board[i][j]);
                
            }
        }
        return true;
    }
}